internal/provider/framework/provider.go

```diff
@@ -0,0 +1,491 @@
+package framework
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	providerfunction "github.com/hashicorp/terraform-provider-azurerm/internal/provider/function"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/resourceproviders"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk/frameworkhelpers"
+)
+
+type azureRmFrameworkProvider struct {
+	V2Provider interface{ Meta() interface{} }
+	ProviderConfig
+}
+
+var _ provider.Provider = &azureRmFrameworkProvider{}
+
+var _ provider.ProviderWithFunctions = &azureRmFrameworkProvider{}
+
+func (p *azureRmFrameworkProvider) Functions(_ context.Context) []func() function.Function {
+	return []func() function.Function{
+		providerfunction.NewNormaliseResourceID,
```

Comment:
Worth keeping these to the same naming convention? e.g.

```suggestion
		providerfunction.NewNormaliseResourceIDFunction,
```